packages/app/src/sessions.ts

```diff
@@ -308,90 +329,133 @@ export function useRRWebEventStream(
-                .split('\n')
-                .map((row: string) => {
-                  try {
-                    const parsed = JSON.parse(row);
-                    linesFetched++;
-                    return parsed;
-                  } catch (e) {
-                    return null;
-                  }
-                })
-                .filter((v: any) => v !== null);
-
-              if (onEvent != null) {
-                parsedRows.forEach(onEvent);
-              } else if (keepPreviousData) {
-                resBuffer.push(...parsedRows);
-              } else {
-                setResults(prevResults => ({
-                  key: resultsKey ?? prevResults.key ?? 'DEFAULT_KEY',
-                  data: [...prevResults.data, ...parsedRows],
-                }));
-              }
-            } else if (event.event === 'end') {
-              onEnd?.();
-
-              if (keepPreviousData) {
-                setResults({
-                  key: resultsKey ?? 'DEFAULT_KEY',
-                  data: resBuffer,
-                });
-              }
-
-              if (linesFetched === 0 || linesFetched < limit) {
-                setHasNextPage(false);
-              }
-            }
-          },
-          onclose() {
-            ctrl.abort();
-
-            setIsFetching(false);
-            setFetchStatus('idle');
-            lastFetchStatusRef.current = 'idle';
-            // if the server closes the connection unexpectedly, retry:
-            // throw new RetriableError();
-          },
-          // onerror(err) {
-          //   if (err instanceof FatalError) {
-          //     throw err; // rethrow to stop the operation
-          //   } else {
-          //     // do nothing to automatically retry. You can also
-          //     // return a specific retry interval here.
-          //   }
-          // },
         },
+        metadata,
       );
 
+      let hostname: string | undefined;
+      let pathname: string | undefined;
+      let username: string | undefined;
+      let password: string | undefined;
+      if (IS_LOCAL_MODE) {
+        const localConnections = getLocalConnections();
+        const fullUrl = new URL(localConnections[0].host);
+        hostname = fullUrl.hostname;
+        pathname = fullUrl.pathname;
+        username = localConnections[0].username;
+        password = localConnections[0].password;
+      }
+
+      // TODO: Change ClickhouseClient class to use this under the hood,
+      // and refactor this to use ClickhouseClient.query
+      const clickhouseClient = createClient({
+        clickhouse_settings: {
+          date_time_output_format: 'iso',
+          wait_end_of_query: 0,
+          cancel_http_readonly_queries_on_client_close: 1,
+        },
+        http_headers: !IS_LOCAL_MODE
+          ? {
+              'x-hyperdx-connection-id': source.connection,
+            }
+          : undefined,
+        url: hostname ?? window.location.origin,
```

Comment:
I got this error while testing the demo sessions. I suspect its due to the validation from the `createClient`
<img width="649" alt="image" src="https://github.com/user-attachments/assets/be054e63-ef17-4030-99fa-bc977d494abd" />

Reply:
I tested demo sessions locally and I got this error
<img width="844" alt="image" src="https://github.com/user-attachments/assets/c6c86fb9-5134-4e84-926e-942aafebecde" />